Sources/ArcGISToolkit/Components/Form/FormInputs/TextInput.swift

```diff
@@ -0,0 +1,223 @@
+// Copyright 2023 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import ArcGIS
+
+/// A view for text input.
+struct TextInput: View {
+    @Environment(\.formElementPadding) var elementPadding
+    
+    /// The model for the ancestral form view.
+    @EnvironmentObject var model: FormViewModel
+    
+    /// The model for the input.
+    @StateObject var inputModel: FormInputModel
+    
+    /// A Boolean value indicating whether or not the field is focused.
+    @FocusState private var isFocused: Bool
+    
+    /// The current text value.
+    @State private var text = ""
+    
+    /// A Boolean value indicating whether placeholder text is shown, thereby indicating the
+    /// presence of a value.
+    ///
+    /// If iOS 16.0 minimum APIs are not supported we use a TextField for single line entry and a
+    /// TextEditor for multiline entry. TextEditors don't have placeholder support so instead we
+    /// replace empty text with the configured placeholder message and adjust the font
+    /// color.
+    ///
+    /// Once iOS 16.0 is the minimum supported platform this property can be removed.
+    @State private var isPlaceholder = false
+    
+    /// The input's parent element.
+    private let element: FieldFormElement
+    
+    /// Creates a view for text input spanning multiple lines.
+    /// - Parameters:
+    ///   - element: The input's parent element.
+    init(element: FieldFormElement) {
+        precondition(
+            element.input is TextAreaFormInput || element.input is TextBoxFormInput,
+            "\(Self.self).\(#function) element's input must be \(TextAreaFormInput.self) or \(TextBoxFormInput.self)."
+        )
+        self.element = element
+        _inputModel = StateObject(
+            wrappedValue: FormInputModel(fieldFormElement: element)
+        )
+    }
+    
+    var body: some View {
+        InputHeader(label: element.label, isRequired: inputModel.isRequired)
+            .padding([.top], elementPadding)
+        if inputModel.isEditable {
+            textField
+        } else {
+            Text(text.isEmpty ? "--" : text)
```

Comment:
Looks a little cramped, so I think it needs the same padding as in `formInputStyle()`. Not sure about adding the minimum frame height.
```suggestion
            Text(text.isEmpty ? "--" : text)
                .padding([.horizontal], 10)
                .padding([.vertical], 5)
```